### PR TITLE
Increase verbosity for unexpected errors

### PR DESCRIPTION
**Pull request by amingilani**
Improves error details instead of just

```
::error::Something went wrong
```

A [GitHub incident](https://www.githubstatus.com/incidents/vts24q167tc8) resulted in that most unhelpful error.

Since GitHub has since fixed the issue, we can't reproduce it. However, in the future, we should hopefully have a better indication of what the problem is.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -159,7 +159,7 @@ async function main() {
 
 // Don't automatically run the main function if we're running a test suite
 if (process.env.NODE_ENV !== "test") {
-  main().catch(() => core.setFailed('Something went wrong'));
+  main().catch((e) => core.setFailed(`Something went wrong. ${e}`));
```
Comment (mwarkentin) [inline index.js:162]: Just curious, but could we just remove the `catch` and let the exception bubble up?

Comment (amingilani) [reply index.js:162]: Good question. Lemme explore how that might work, I know this line explicitly fails the action with [`core.setFailed`](https://docs.github.com/en/free-pro-team@latest/actions/creating-actions/setting-exit-codes-for-actions), lemme test out what happens if it bubbles up.

Comment (mwarkentin) [reply index.js:162]: Cool, sounds like that might be the correct way to handle things then. Let me know what you find!

Comment (amingilani) [reply index.js:162]: I'm using [`act`](https://github.com/nektos/act) to test this out locally, which is not perfect, but as expected the action succeeds despite the error. Additionally, Node.js complains about me breaking promise aka `UnhandledPromiseRejectionWarning`

Full log:

```
[Migration Check/Check migration files] 🚀  Start image=node:12.6-buster-slim
[Migration Check/Check migration files]   🐳  docker run image=node:12.6-buster-slim entrypoint=["/usr/bin/tail" "-f" "/dev/null"] cmd=[]
[Migration Check/Check migration files] ⭐  Run Check migration files step
[Migration Check/Check migration files]   ☁  git clone 'https://github.com/waveaccounting/migbot-github-action' # ref=6c8a2c870af849c63f9b07f15ac987658180c874
[Migration Check/Check migration files]   🐳  docker cp src=/Users/agilani/.cache/act/waveaccounting-migbot-github-action@6c8a2c870af849c63f9b07f15ac987658180c874 dst=/actions/
| (node:12) UnhandledPromiseRejectionWarning: Error: The sky is failing!
|     at main (/actions/waveaccounting-migbot-github-action@6c8a2c870af849c63f9b07f15ac987658180c874/index.js:48:9)
|     at Object.<anonymous> (/actions/waveaccounting-migbot-github-action@6c8a2c870af849c63f9b07f15ac987658180c874/index.js:164:3)
|     at Module._compile (internal/modules/cjs/loader.js:776:30)
|     at Object.Module._extensions..js (internal/modules/cjs/loader.js:787:10)
|     at Module.load (internal/modules/cjs/loader.js:643:32)
|     at Function.Module._load (internal/modules/cjs/loader.js:556:12)
|     at Function.Module.runMain (internal/modules/cjs/loader.js:839:10)
|     at internal/main/run_main_module.js:17:11
| (node:12) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
| (node:12) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
[Migration Check/Check migration files]   ✅  Success - Check migration files step
```

**Review by mwarkentin — APPROVED** [review]
LGTM

**Pull request merged**